src/ui/TableOfContent.tsx

```diff
@@ -9,33 +9,151 @@ import {
+    color: isActive ? activeColor : color,
+    borderBottom: `1px solid`,
+    borderColor: borderColor,
+    py: 3,
+    _hover,
+    _focus,
+  };
+
+  return (
+    <MenuItem as={Link} href={href} {...styles} {...rest}>
+      <RenderHtml title={title ?? ''} />
+    </MenuItem>
+  );
+};
+
 export default function TableOfContent({
   navigator,
   manifest,
   readerState,
 }: {
-  navigator: Navigator;
+  navigator: PdfNavigator | HtmlNavigator;
   manifest: WebpubManifest;
-  readerState: ReaderState;
+  readerState: ReaderState | PdfReaderState;
 }): React.ReactElement {
   const [isOpen, setIsOpen] = useState(false);
-  const tocLinkHandler: React.MouseEventHandler<HTMLButtonElement> = (evt) => {
+  const [outline, setOutline] = useState<JSX.Element[]>([]);
+
+  const tocLinkHandler = (
+    evt: React.MouseEvent<Element, MouseEvent>,
+    item?: PDFTreeNode
+  ) => {
     evt.preventDefault();
-    const href = evt.currentTarget.getAttribute('href');
-    if (!href) {
-      console.warn('TOC Link clicked without an href');
-      return;
+    const href: string | null = evt.currentTarget.getAttribute('href');
+    console.log('href', href);
+    if (href && href !== '#') {
+      navigator.goToPage(href);
+      setIsOpen(false);
+    } else if (item && item.dest) {
+      console.log('item', item);
+      navigator.goToPage(item.dest);
+      setIsOpen(false);
     }
-    navigator.goToPage(href);
-    setIsOpen(false);
   };
 
+  const pdfResource = (readerState as PdfReaderState).pdf;
+
+  const pdfToc =
+    pdfResource &&
+    manifest &&
+    manifest.resources &&
+    manifest.resources.length === 1;
+
+  if (pdfToc) {
+    const getPdfOutline = (outline: PDFTreeNode[] | undefined) => {
+      if (!outline)
+        throw new Error(
+          'Cannot call getPdfOutline when there is no PDF Reader'
+        );
+      return outline.map((content: PDFTreeNode) => (
+        <React.Fragment key={content.title}>
+          <TocItem
+            href="#"
+            title={content.title}
+            isActive={false}
+            onClick={(e) => {
+              tocLinkHandler(e, content);
+            }}
+          />
+          {content.items &&
+            content.items.map((subLink: PDFTreeNode) => (
+              <TocItem
+                key={subLink.title}
+                href="#"
+                title={subLink.title}
+                isActive={false}
+                onClick={(e) => tocLinkHandler(e, subLink)}
+                pl={10}
+              />
+            ))}
+        </React.Fragment>
+      ));
+    };
+
+    const getOutline = async () => {
+      if (pdfResource) {
+        const outline: PDFTreeNode[] = await pdfResource.getOutline();
+        const pdfOutline = getPdfOutline(outline);
+        setOutline(pdfOutline);
+      } else {
+        throw new Error('trying to get state');
+      }
+    };
+
+    if (outline.length === 0) {
+      getOutline();
+    }
+  }
+
```

Comment:
Question: 

Is it possible in `usePdfReader` to process the pdf outline into a readium-style `toc`? Then we could just add it to the webpub manifest we return and render it normally using this component